wordpress-mobile/WordPress-Android
# Jepack App (Basic): UI Tests for Screenshots

Parent #14634 

This PR adds Jetpack App Screenshots support, specifically new UI tests were added for:
- Site
- Activity Log
- Scan
- Backup Download
- Stats

It is best to review this PR by commit. There are a lot mock `.json` file changes, so the file count is high, as well as the number of lines. 

The way I approached this work was as follows:
- Executed WP connected tests locally before any coding changes to check for all passing
- Executed WPScreenshotTest.wPScreenShotTest before any coding changes to check for all passing
- Fixed failing WP Connected and WPScreenshotTest tests by updating `blocks.json`
- Added JPScreenshotTest 1x1, updating/adding any mock .json files as I encountered errors
- The test section that required the most new .json files was Navigate Sites 
- The test section that required the most updates to .json files was Stats.
- As part of the Stats update, I refactored a good portion of the .json files to be more generic so that could be used for other sites. This mostly included changing either
            -  `urlPath` to `urlPattern` and replacing the hardcoded site id with `.*`. 
            OR
            -  `urlPath` to `urlPattern` and replacing the hardcoded site id with `([0-9]+)`. 
- The code changes are limited to three classes:
    - `WPScreenshotTest`
    - `JPScreenshotTest`
    - `MySitePages.java`
- I added an if statement to both `WPScreenshotTest` & `JPScreenshotTest`, so that they only run when executing against the proper build. `WPScreenshotTest` for WordPress app builds and `JPScreenshotTest` for Jetpack app builds. I found that they both run when connected tests are running and will both fail when run against the wrong build. If you have a better suggestion, I am all ears.
- I had considered creating a androidTestJetpack, but the infrastructure is not set up for it in CI and out of caution for scope, I did not feel it was prudent to start down that road 2 weeks before release. This can be revisited after release.
- For a sanity check, connected tests, WPScreenshotTest and JPScreenshotTest were executed throughout the entire development process, as well as after the final commit.
- There may be changes needed based on design @osullivanchris . All data for stats based on data that is preexisiting. I will cover any design changes in a separate PR because I believe it may include changes to many .json files.


My Site | Activity Log | Jetpack Scan | Jetpack Backup | Stats
-- | -- | -- | -- | --
![My Site](https://user-images.githubusercontent.com/506707/119277374-3c67a200-bbed-11eb-8ac1-d36e77e3a1e2.png) | ![Activity Log](https://user-images.githubusercontent.com/506707/119277366-3245a380-bbed-11eb-90d3-901b75be6335.png) | ![Scan](https://user-images.githubusercontent.com/506707/119277349-1a6e1f80-bbed-11eb-816e-a718752dd9cc.png) | ![Backup](https://user-images.githubusercontent.com/506707/119277356-23f78780-bbed-11eb-9747-7f49e1aeae53.png) | ![Stats](https://user-images.githubusercontent.com/506707/119277342-117d4e00-bbed-11eb-96da-1e3c9f613e61.png)


To test:
- Check out this branch
- Select a Jetpack debug variant
- Run `JPScreenshotTest.jPScreenshotTest`
- 1 test should pass
------------------------
- Change the build to a WordPress variant
- Run `WPScreenshotTest.wPScreenshotTest`
- 1 test should pass

## Regression Notes
1. Potential unintended areas of impact
WPScreenshots and/or WP Connected tests

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Ran WP connected tests after each JP change
Ran WPScreenshot tests after each JP change

3. What automated tests I added (or what prevented me from doing so)
JPScreenshot

PR submission checklist:
